adoc/MAIN.release-notes.adoc

```diff
@@ -19,6 +19,82 @@ Release notes usually only list changes that happened between two subsequent rel
 Certain important entries from the release notes documents of previous product versions may be repeated.
 To make such entries easier to identify, they contain a note to that effect.
 
+== Changes in 4.1.2
+
+{productname} can now be deployed with *{tf} 0.12*. All details of the new version
+can be found in the link:https://www.hashicorp.com/blog/terraform-0-1-2-preview/[HashiCorp Documentation].
+The official website for the {tf} 0.12 upgrade is https://www.terraform.io/upgrade-guides/0-12.html.
+
+=== Required Actions
+
+==== Upgrade {tf} Files and State
+
+In order to seamlessly switch to {tf} 0.12 you need to make sure that:
+
+* all files follow the new syntax for the HashiCorp Configuration Language included in {tf} 0.12;
+* all boolean values are ``true`` or ``false`` and *not* 0 or 1;
+* all variables are explicitly declared;
+* all dependencies are explicitly declared to reach the expected behavior.
+
+==== Recommended Procedure
+
+If you can tear down your existing cluster, do delete your cluster
+before upgrading to {tf} 0.12. After that follow our documentation to create a new cluster.
+That will lead to the cleanest upgrade result.
+
+If you are using {tf} 0.11 and you cannot tear down your cluster, you will
+need to update your {tf} files (and states) in place for {tf} 0.12.
+
+To do this, enter your {tf} files/state folder and:
+
+* Migrate {tf} files with the automatic migration tool by running ``terraform 0.12upgrade``.
+* Run the extra operations for in-place upgrade, which follow just below.
+
+==== Extra Operations for In-place Upgrade of OpenStack {tf} Files
+
+* Replace any boolean values written as a number with ``false``/``true``. For example, replace
+  ``default = 0`` with ``default = false`` in ``openstack/variables.tf``.
```

Comment:
It would be more transparent, if we said which specific values are boolean and should be changed. We tell our users to deploy on openstack using `openstack/terraform.tfvars.example`, so which values would they change specifically if the deploy using that template?

Reply:
We can't presume things won't be modified, and that's why we are giving the instructions instead. 
dnsentry, cpi_enable, and *_vol_enabled are the ones in variables.tf by default. I can add extra text listing those, and mentioning it's a non exhaustive list?

Reply:
Yes, I would remove the `default` example and say they should change those three specifically plus any other ones they might have added.

Reply:
OK.